pkgs/os-specific/darwin/cctools/port.nix

```diff
@@ -12,7 +12,7 @@ let
     "${targetPlatform.config}-";
 in
 
-assert targetPlatform.isDarwin;
+assert stdenv.isDarwin;
```

Comment:
This could potentially also be the reason for the incorrect package set on darwin.